### PR TITLE
fix-284-changie-next-v123

### LINKED ISSUE #284: changie next v1.2.3
**Description**
Maybe I am missing something, but the behavior of the following command is
a bit counter-intuitive (accepted, but not really documented):

1. Form 1:
```
changie next v.1.2.3
v1.2.3
```
2. I would understand the meaning of `changie next patch v1.2.3` (unsupported) or similar command.
3. I would understand the meaning of `changie next v1.2.3` to some automatically determined minor/patch/major based on the content of some unreleased changie tickets (none of this happens).

But the current behavior is a puzzle to me. 

Unless there is a specific rationale for it, I would prefer either of the following fixes:
1. Form (1) returns an error (expect patch|minor|major)
2. Support form (2) (extra arg)
3. Support form (3) (extra complexity, some automagical/under-the-hood stuff is happening, maybe not a good idea).

**Reproduction Steps**
1. Run '...'
1. Edit config '...'
1. Then run '...'
1. See error

**What happened**
What did you see happen?

**Expected behavior**
What did you expect to happen?

**Additional context**

### PR DESCRIPTION
**Pull request by fredbi**
* fixes #284
    
    Signed-off-by: Frederic BIDON <fredbi@yahoo.com>

**Check the following**
- [ ] Keep 100% code coverage
- [ ] Be properly formatted
- [ ] Documentation changes are included
- [ ] Include a change file if expected

**Additional context**
Any additional info that might help get your pull request merged.

### CODE REVIEW

File: cmd/next.go [hunk 1]
```diff
@@ -61,6 +62,10 @@ func nextPipeline(afs afero.Afero, writer io.Writer, part string, prerelease, me
 		return err
 	}
 
+	if part != "patch" && part != "minor" && part != "major" {
```
Comment (miniscruff) [inline cmd/next.go:65]: I would use a `map[string]struct{}` for readability

Comment (miniscruff) [inline cmd/next.go:65]: I actually think we should create a func for the command.Args above to validate we get 1 parameter and it is part of the valid options. https://github.com/spf13/cobra/blob/5b2b9e9f61d36ccb66167301f76a2292c3729855/args.go#L8
Up on line 25, https://github.com/miniscruff/changie/pull/287/files#diff-c2a7c617a6418a3f91c597ddc318504bf0ec9fec0d4314f78ab67b34e14f3230R25

**Review by miniscruff — CHANGES_REQUESTED** [review]

File: cmd/next.go (hunk 1 above)
Comment (fredbi) [reply cmd/next.go]: ok done. Moved test case to e2e.

Comment (fredbi) [reply cmd/next.go]: see below: used cobra lib now.

File: .changes/unreleased/Fixed-20220315-105032.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+kind: Fixed
+body: 'fix(cli): changie next enforces arg to be patch|minor|major'
```
Comment (miniscruff) [inline .changes/unreleased/Fixed-20220315-105032.yaml:2]: No fix cli necessary I would just say
```suggestion
body: 'next command enforces argument to be major, minor or patch'
```

File: cmd/next.go (hunk 1 above)
Comment (miniscruff) [reply cmd/next.go]: Oh nice I actually didn't see that option, that is better than I expected

**Review by miniscruff — CHANGES_REQUESTED** [review]

**Comment by miniscruff** [comment]
I will merge later today and adjust the changie fragment if you are busy. Then I can do a minor release with all your suggestions and fixes.

**Comment by fredbi** [comment]
@miniscruff  ok thanks

**Pull request merged**